spark/hbase-spark/src/test/scala/org/apache/hadoop/hbase/spark/DefaultSourceSuite.scala

```diff
@@ -1359,4 +1359,12 @@ class DefaultSourceSuite
         }
     }
   }
+
+  test("NPE should not be thrown even when HBaseContext was never initialized") {
+    LatestHBaseContextCache.latest = null
+    val catalog = s"""{"table":{"namespace":"default", "name":"table"},
+                     | "rowkey":"key",
+                     | "columns":{"c1":{"cf":"d", "col":"c1", "type":"string"}}}""".stripMargin
+    HBaseRelation(Map(HBaseTableCatalog.tableCatalog -> catalog), None)(sqlContext)
```

Comment:
Please consider adding a try-catch block / assertion here for NPE. We can fail if we get an NPE

Reply:
```suggestion
    try {
      HBaseRelation(Map(HBaseTableCatalog.tableCatalog -> catalog), None)(sqlContext)
    } catch {
      case _: NullPointerException => fail("Unexpected NPE thrown")
    }
```